test/integration/webhook/v1alpha1/workload_test.go

```diff
@@ -70,3 +70,76 @@ var _ = ginkgo.Describe("Workload defaulting webhook", func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Workload validating webhook", func() {
+	ginkgo.Context("When creating a Workload", func() {
+		ginkgo.It("Should validate spec.podSets.count", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload2", ns.Name).Obj()
+			workload.Spec.PodSets = []v1alpha1.PodSet{
+				{
+					Count: -1,
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "test",
+								Image: "fake-image",
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(err.Error()).Should(gomega.ContainSubstring("must be greater than 0"))
+		})
+
+		ginkgo.It("Should validate spec.priorityClassName", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload3", ns.Name).Obj()
+			workload.Spec.PriorityClassName = "invalid_class"
+			err := k8sClient.Create(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(err.Error()).Should(gomega.ContainSubstring("a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters"))
+		})
+	})
+
+	ginkgo.Context("When updating a Workload", func() {
+		ginkgo.It("Should validate spec.podSet.count", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload3", ns.Name).Obj()
+			workload.Spec.PodSets = []v1alpha1.PodSet{
```

Comment:
I wonder whether we really need this as `MakeWorkload()` already contains a qualified podset.

Reply:
We need to update the field `spec.podSet[0].count` later in the test, I think we could set this field in advance.

Reply:
I mean MakeWorkload() already created a podSet in advanced https://github.com/kubernetes-sigs/kueue/blob/99683618876a004f3332fb99cae1e3a9bf6f2124/pkg/util/testing/wrappers.go#L141-L163